editors/code/src/client.ts

```diff
@@ -51,12 +51,24 @@ export function createClient(serverPath: string, cwd: string, extraEnv: Env): lc
         'Rust Analyzer Language Server Trace',
     );
 
+    const workspace:lc.WorkspaceMiddleware = {
+        willRenameFiles:function<P extends vscode.FileRenameEvent,R extends Thenable<vscode.WorkspaceEdit | null | undefined>>(this: void, data: P, next:(data: P) => R ){
+            // why add this function rather than default:
+            // 1. change `url` parameter to happy format for url crate. (folder should end with '/')
+            // 2. filter some change in here.
+            //     2.1 rename from or to `mod.rs` should be special. 
+            //     2.2 not all folder change should be cared, only those have files with ".rs" postfix.
+            return next(data);
```

Comment:
I think these things can be handled server side. I believe the dart language server has some good examples.

Reply:
Thanks for the suggestion!
I have no idea which side is a better place to handle this. In the new change I filter out "mod" in server side.
Is there any other suggestions?

Reply:
I think that all of this should live in the server-side logic. We try to keep everything there since there are clients beside vscode.